src/coreclr/jit/importer.cpp

```diff
@@ -4515,13 +4515,54 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             case NI_System_Math_Log:
             case NI_System_Math_Log2:
             case NI_System_Math_Log10:
-#ifdef TARGET_ARM64
+#if defined TARGET_ARM64
             // ARM64 has fmax/fmin which are IEEE754:2019 minimum/maximum compatible
             // TODO-XARCH-CQ: Enable this for XARCH when one of the arguments is a constant
             // so we can then emit maxss/minss and avoid NaN/-0.0 handling
             case NI_System_Math_Max:
             case NI_System_Math_Min:
 #endif
+
+#if defined FEATURE_HW_INTRINSICS && defined TARGET_XARCH
+            case NI_System_Math_Max:
+            case NI_System_Math_Min:
+            {
+                assert(varTypeIsFloating(callType));
+
+                if (sig->numArgs == 2 && (impStackTop().val->IsCnsFltOrDbl() || impStackTop(1).val->IsCnsFltOrDbl()))
+                {
+                    GenTree* op2 = impPopStack().val;
+                    GenTree* op1 = impPopStack().val;
+
+                    unsigned int insSimdSize = (callType == TYP_DOUBLE) ? 32 : 16;
```

Comment:
Why would we need 32-byte SIMD instructions for `TYP_DOUBLE`?

Reply:
I thought that double needed this type of SIMD. I was wrong, wasn't I?

Reply:
`TYP_DOUBLE` and `TYP_FLOAT` can actually both operate directly as scalar, but both will return `TYP_SIMD16`.